### PR TITLE
CLDR 46 Update - Country, langname

### PR DESCRIPTION
**Pull request by gouniLee**
### Checklist

* [ ] At least one test case is included for this feature or bug fix.
* [ ] `ReleaseNotes` has added or is not needed.
* [ ] This PR has passed all of the test cases on `Nodejs` and `Chrome Browser`.
* [ ] This PR has passed all of the test cases on `QT/QML`.
* [ ] This is an API breaking change.
* [ ] Requires a major version change.

### Issue Resolved / Feature Added
* Update ctrynames.js, ctryreverse.json and regionnames.json to cldr v46
  *  In CLDR v46, the HK region information has been removed from `common/subdivisions/en.xml`, so the tool code has been updated to preserve the data.
* Update langname.jf to cldr  v46

### Resolution
[//]: # (Does the code work as intended?)
[//]: # (What is the impact of this change and *why* was it made?)

### Links
[//]: # (Related issues, references)

### CODE REVIEW

File: js/data/locale/en/regionnames.json [hunk 1]
```diff
@@ -8127,12 +8179,6 @@
             "name": "Upper Takutu-Upper Essequibo"
         }
     ],
-    "HK": [
```
Comment (ehoogerbeets) [inline js/data/locale/en/regionnames.json:8130]: Why is this deleted?

Comment (gouniLee) [reply js/data/locale/en/regionnames.json:8130]: To generated the data, The tool uses the xml file (common/subdivisions/*.xml).  HK part is deleted from the en.xml file. 
`<subdivision type="HK">Hong Kong SAR China</subdivision>`

**Comment by gouniLee** [comment]
I update the tool code and re-generated the data. Please check them again.

**Review by ehoogerbeets — APPROVED** [review]
